### PR TITLE
test(e2e): add challenge timeout fail case

### PR DESCRIPTION
**Pull request by kangsorang**
# Description

Add e2e test about challenge timeout fail case.

### CODE REVIEW

File: e2e/actions/l2_challenger_test.go [hunk 1]
```diff
@@ -66,7 +66,90 @@ interaction:
...
+			break interaction
+		}
+
+		// Check whether the submission was successful
+		rt.receipt, err = rt.miner.EthClient().TransactionReceipt(rt.t.Ctx(), rt.txHash)
+		require.NoError(rt.t, err)
+		require.Equal(rt.t, types.ReceiptStatusSuccessful, rt.receipt.Status, "failed to progress interactive proof")
+	}
+
+	// Check the status of challenge is StatusChallengerTimeout(3)
```
Comment (0xHansLee) [inline e2e/actions/l2_challenger_test.go:104]: The status of challenge is different.
In comment `ChallengerTimeout` but `AsserterTimeout` in assertion statement.

**Review by 0xHansLee — COMMENTED** [review]
It looks like we need to change our test name:
- `TestChallenge`
- `TestChallengeAsserterBisectTimeout`
- `TestChallengeChallengerBisectTimeout`
- `TestChallengeChallengerProvingTimeout`
- `TestChallengeInvalidProof`
wdyt?

File: e2e/actions/l2_challenger_test.go [hunk 2]
```diff
@@ -66,7 +66,90 @@ interaction:
...
+			rt.txHash = rt.challenger.ActProveFault(rt.t, rt.outputIndex, false)
+			rt.miner.includeL1Block(rt.t, rt.challenger.address)
+		default:
+			break interaction
+		}
+
+		// Check whether the submission was successful
+		rt.receipt, err = rt.miner.EthClient().TransactionReceipt(rt.t.Ctx(), rt.txHash)
+		require.NoError(rt.t, err)
+		require.Equal(rt.t, types.ReceiptStatusSuccessful, rt.receipt.Status, "failed to progress interactive proof")
```
Comment (chokobole) [inline e2e/actions/l2_challenger_test.go:101]:
```suggestion
		require.Equal(rt.t, types.ReceiptStatusSuccessful, rt.receipt.Status, "failed to progress interactive fault proof")
```

File: e2e/actions/l2_challenger_test.go [hunk 3]
```diff
@@ -66,7 +66,90 @@ interaction:
...
+			rt.txHash = rt.validator.ActBisect(rt.t, rt.outputIndex)
+			rt.miner.includeL1Block(rt.t, rt.validator.address)
+		default:
+			break interaction
+		}
+
+		// Check whether the submission was successful
+		rt.receipt, err = rt.miner.EthClient().TransactionReceipt(rt.t.Ctx(), rt.txHash)
+		require.NoError(rt.t, err)
+		require.Equal(rt.t, types.ReceiptStatusSuccessful, rt.receipt.Status, "failed to progress interactive proof")
```
Comment (chokobole) [inline e2e/actions/l2_challenger_test.go:143]:
```suggestion
		require.Equal(rt.t, types.ReceiptStatusSuccessful, rt.receipt.Status, "failed to progress interactive fault proof")
```

File: e2e/e2eutils/setup.go [hunk 4]
```diff
@@ -104,7 +104,7 @@ func MakeDeployParams(t require.TestingT, tp *TestParams) *DeployParams {
 		ColosseumProvingTimeout:   480,
 		ColosseumDummyHash:        common.HexToHash("0x6cf9919fd9dfe923ed2f2e4d980d677a88d17c74f8f6604ffac1512ff306e760"),
 		ColosseumMaxTxs:           25,
-		ColosseumSegmentsLengths:  "3,4",
+		ColosseumSegmentsLengths:  "2,2,3,4",
```
Comment (chokobole) [inline e2e/e2eutils/setup.go:107]: Is it a change to be related to this commit?

File: e2e/actions/l2_challenger_test.go [hunk 5]
```diff
@@ -66,7 +66,90 @@ interaction:
...
+		status, err := rt.colosseumContract.GetStatus(nil, rt.outputIndex)
+		require.NoError(rt.t, err)
+
+		switch status {
+		case chal.StatusAsserterTurn:
+			// do nothing to trigger asserter timeout
+			rt.miner.ActEmptyBlock(rt.t)
+		case chal.StatusChallengerTurn:
+			rt.txHash = rt.challenger.ActProveFault(rt.t, rt.outputIndex, false)
+			rt.miner.includeL1Block(rt.t, rt.challenger.address)
```
[the remark this answers is not in the document]
Comment (kangsorang) [reply e2e/actions/l2_challenger_test.go:93]: Correct!

File: e2e/e2eutils/setup.go (hunk 4 above)
Comment (kangsorang) [reply e2e/e2eutils/setup.go:107]: Yes, Increased the number of turns to 4 for the test challenger's bisect timeout.

**Review by chokobole — APPROVED** [review]
LGTM

File: e2e/actions/l2_challenger_test.go [hunk 6]
```diff
@@ -66,7 +66,94 @@ interaction:
...
+		switch status {
+		case chal.StatusAsserterTurn:
+			// do nothing to trigger asserter timeout
+			rt.miner.ActEmptyBlock(rt.t)
+		case chal.StatusChallengerTurn:
+			// call bisect by challenger
+			rt.txHash = rt.challenger.ActBisect(rt.t, rt.outputIndex)
+			rt.miner.includeL1Block(rt.t, rt.challenger.address)
+		case chal.StatusAsserterTimeout:
+			rt.txHash = rt.challenger.ActProveFault(rt.t, rt.outputIndex, false)
```
Comment (0xHansLee) [inline e2e/actions/l2_challenger_test.go:96]: This should be `true`, which make test succeeds
```suggestion
			rt.txHash = rt.challenger.ActProveFault(rt.t, rt.outputIndex, true)
```

Comment (kangsorang) [reply e2e/actions/l2_challenger_test.go]: @0xHansLee  ur right. thx!

**Review by 0xHansLee — APPROVED** [review]
lgtm

File: e2e/actions/l2_challenger_test.go [hunk 7]
```diff
@@ -66,7 +66,94 @@ interaction:
...
+		require.NoError(rt.t, err)
+
+		switch status {
+		case chal.StatusAsserterTurn:
+			// do nothing to trigger asserter timeout
+			rt.miner.ActEmptyBlock(rt.t)
+		case chal.StatusChallengerTurn:
+			// call bisect by challenger
+			rt.txHash = rt.challenger.ActBisect(rt.t, rt.outputIndex)
+			rt.miner.includeL1Block(rt.t, rt.challenger.address)
```
Comment (seolaoh) [inline e2e/actions/l2_challenger_test.go:94]: This code block is unnecessary. If asserter timed out, challenger turn never comes.

File: e2e/actions/l2_challenger_test.go [hunk 8]
```diff
@@ -66,7 +66,94 @@ interaction:
...
+		case chal.StatusAsserterTimeout:
+			rt.txHash = rt.challenger.ActProveFault(rt.t, rt.outputIndex, true)
+			rt.miner.includeL1Block(rt.t, rt.challenger.address)
+		default:
```
Comment (seolaoh) [inline e2e/actions/l2_challenger_test.go:98]: I think we need to handle `StatusProven` also here, and the final status of challenge should be `StatusApproved` with malicious asserter, honest challenger and honest guardian.

File: e2e/actions/l2_challenger_test.go [hunk 9]
```diff
@@ -66,7 +66,99 @@ interaction:
...
+interaction:
+	for {
+		status, err := rt.colosseumContract.GetStatus(nil, rt.outputIndex)
+		require.NoError(rt.t, err)
+
+		switch status {
+		case chal.StatusAsserterTurn:
+			// do nothing to trigger asserter timeout
+			rt.miner.ActEmptyBlock(rt.t)
+		case chal.StatusAsserterTimeout, chal.StatusReadyToProve:
```
Comment (seolaoh) [inline e2e/actions/l2_challenger_test.go:92]: In this test case, `StatusReadyToProve` should not come. If `StatusReadyToProve` comes by a bug, this test case should be failed.
```suggestion
		case chal.StatusAsserterTimeout:
```

**Review by seolaoh — APPROVED** [review]
LGTM

**Pull request merged**